### ISSUE TITLE
Request batching with tensorflow model doesn't work

### ISSUE THREAD

**Issue by hifazibm** [kserve/kserve#2114]
/kind bug

When request batch parameters are set for tensorflow model, the inference service does not generate pods. Same issue is also observed for pytorch model.

**Environment:**

- **KFServing Version: 0.8 (Raw Deployment Mode)**
- Kubernetes version:
 kubectl version
Client Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.4+82e7b55", GitCommit:"82e7b55", GitTreeState:"clean", BuildDate:"2019-05-20T00:17:47Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"b09a9ce3", GitTreeState:"clean", BuildDate:"2022-03-11T12:10:16Z", GoVersion:"go1.16.14", Compiler:"gc", Platform:"linux/amd64"}

**1. Yaml File**
```
apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: "flower-sample"
spec:
  predictor:
    minReplicas: 1
    timeout: 60
    batcher:
      maxBatchSize: 32
      maxLatency: 500
    model:
      modelFormat:
        name: tensorflow
      storageUri: "gs://kfserving-samples/models/tensorflow/flowers"
```
**2. oc describe isvc/flower-sample**
Status:
 Components:
  Predictor:
   Latest Created Revision: 1
 Conditions:
  Last Transition Time: 2022-03-21T21:28:52Z
  Reason:        Predictor ingress not created
  Status:        False
  Type:         IngressReady
  Last Transition Time: 2022-03-21T21:28:52Z
  Message:        Deployment does not have minimum availability.
  Reason:        MinimumReplicasUnavailable
  Status:        False
  Type:         PredictorReady
  Last Transition Time: 2022-03-21T21:28:52Z
  Reason:        Predictor ingress not created
  Status:        False
  Type:         Ready
Events:          <none>

**3. kserve manager logs**
2022/03/21 21:46:52 http: panic serving 10.42.0.1:60774: runtime error: index out of range [0] with length 0
goroutine 12698 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1802 +0xb9
panic({0x1965d40, 0xc000c62c90})
	/usr/local/go/src/runtime/panic.go:1047 +0x266
[github.com/kserve/kserve/pkg/webhook/admission/pod.(*AgentInjector).InjectAgent(0xc00099c8f0](http://github.com/kserve/kserve/pkg/webhook/admission/pod.(*AgentInjector).InjectAgent(0xc00099c8f0), 0xc0009d0800)
	/go/src/github.com/kserve/kserve/pkg/webhook/admission/pod/agent_injector.go:213 +0x1e6f
[github.com/kserve/kserve/pkg/webhook/admission/pod.(*Mutator).mutate(0xc0006b9630](http://github.com/kserve/kserve/pkg/webhook/admission/pod.(*Mutator).mutate(0xc0006b9630), 0x1d61b28, 0xc000040338)
	/go/src/github.com/kserve/kserve/pkg/webhook/admission/pod/mutator.go:121 +0x1ff
[github.com/kserve/kserve/pkg/webhook/admission/pod.(*Mutator).Handle(_](http://github.com/kserve/kserve/pkg/webhook/admission/pod.(*Mutator).Handle(_), {_, _}, {{{0xc000c24d80, 0x24}, {{0x0, 0x0}, {0xc0011075f0, 0x2}, {0xc0011075f2, ...}}, ...}})
	/go/src/github.com/kserve/kserve/pkg/webhook/admission/pod/mutator.go:65 +0x372
[sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle(_](http://sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle(_), {_, _}, {{{0xc000c24d80, 0x24}, {{0x0, 0x0}, {0xc0011075f0, 0x2}, {0xc0011075f2, ...}}, ...}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/webhook/admission/webhook.go:146 +0xa2
[sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP(0xc0006ba9f0](http://sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP(0xc0006ba9f0), {0x7f6fd8a48220, 0xc0009c45f0}, 0xc00100c700)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/webhook/admission/http.go:99 +0xef0
[github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerInFlight.func1({0x7f6fd8a48220](http://github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerInFlight.func1(%7B0x7f6fd8a48220), 0xc0009c45f0}, 0xc0009c45f0)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:40 +0xd4
net/http.HandlerFunc.ServeHTTP(0x1d5b720, {0x7f6fd8a48220, 0xc0009c45f0}, 0x7f6fffa685b8)
	/usr/local/go/src/net/http/server.go:2047 +0x2f
[github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1({0x1d5b720](http://github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1(%7B0x1d5b720), 0xc00079a2a0}, 0xc00100c700)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:101 +0x92
net/http.HandlerFunc.ServeHTTP(0xc00083fd80, {0x1d5b720, 0xc00079a2a0}, 0xc0007b2000)
	/usr/local/go/src/net/http/server.go:2047 +0x2f
[github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2({0x1d5b720](http://github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2(%7B0x1d5b720), 0xc00079a2a0}, 0xc00100c700)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:76 +0xa2
net/http.HandlerFunc.ServeHTTP(0x0, {0x1d5b720, 0xc00079a2a0}, 0xc00079a2a0)
	/usr/local/go/src/net/http/server.go:2047 +0x2f
net/http.(*ServeMux).ServeHTTP(0xc000c24d32, {0x1d5b720, 0xc00079a2a0}, 0xc00100c700)
	/usr/local/go/src/net/http/server.go:2425 +0x149
net/http.serverHandler.ServeHTTP({0x1d4de90}, {0x1d5b720, 0xc00079a2a0}, 0xc00100c700)
	/usr/local/go/src/net/http/server.go:2879 +0x43b
net/http.(*conn).serve(0xc000f04320, {0x1d61b98, 0xc0000ad5f0})
	/usr/local/go/src/net/http/server.go:1930 +0xb08
created by net/http.(*Server).Serve

@yuzisun if you could please provide a temporary fix it would be great, need it to demonstrate KServe features for client demo. Thanks.

**Comment by andyi2it** [comment]
Hi @hifazibm , am currently looking at the issue, can you format your yamls and update here. Please use backtics to put yaml in code block, to ensure there are no issues wrt yaml syntax.

**Comment by yuzisun** [comment]
@andyi2it Does container port always exist [here](https://github.com/kserve/kserve/blob/master/pkg/webhook/admission/pod/agent_injector.go#L213)? Might be only true for serverless mode but not raw deployment mode

**Comment by hifazibm** [comment]
Thanks @andyi2it  i updated the yaml file format

**Comment by hifazibm** [comment]
Hi @andyi2it  thanks for the fix, is there any way I can test the fix on my environment (V0.8) by updating KServe.yaml file.
